torch/_inductor/utils.py

```diff
@@ -709,6 +709,21 @@ def cache_property_on_self(fn: Callable[P, RV]) -> CachedMethod[P, RV]:
     return cache_on_self(fn)
 
 
+def cache_on_self_and_args(
+    fn: Callable[Concatenate[Any, P], RV],
+) -> Callable[Concatenate[Any, P], RV]:
+    cache = {}
+
+    @functools.wraps(fn)
+    def wrapper(self: Any, *args: P.args, **kwargs: P.kwargs) -> RV:
+        key = (id(self), args, tuple(sorted(kwargs.items())))
+        if key not in cache:
+            cache[key] = fn(self, *args, **kwargs)
```

Comment:
Looking at how `cache_on_self` was implemented, I think we should do something similar here to further improve the performance.